tests/Behat/WebtoolsContext.php

```diff
@@ -83,8 +89,29 @@ public function assertWebtoolsWidgetExist(string $widget_type, string $title): v
     // Run the escaping on the Json data.
     $snippet = Json::encode(Json::decode($media->get('oe_media_webtools')->value));
     // Escape \ and ' for the xpath expression.
-    $xpath = "//script[@type='application/json'][.='" . addcslashes($snippet, '\\\'') . "']";
-    $this->assertSession()->elementsCount('xpath', $xpath, 1);
+    $xpath_query = "//script[@type='application/json'][.='" . addcslashes($snippet, '\\\'') . "']";
+    // Assert presence of webtools JSON with enabled javascript.
+    if ($this->browserSupportsJavaScript()) {
+      // Receive unchanged HTML of the current page by ajax.
+      $script = <<<JS
+        (function(window) {
+          var xhr = new XMLHttpRequest();
+          xhr.open('GET', window.location.href, false);
+          xhr.send(null);
+          return xhr.responseText;
+        })(window)
+JS;
+
+      $raw_html = $this->getSession()->evaluateScript($script);
+      $doc = new \DOMDocument();
+      libxml_use_internal_errors(TRUE);
+      $doc->loadHTML($raw_html);
```

Comment:
Let's follow what core does in `\Drupal\Component\Utility\Html::load()`:
```suggestion
      $doc = new \DOMDocument();
      @$doc->loadHTML($raw_html);
```